### ISSUE TITLE
"Resolving wheels disallowed" error when using `--lock` and `--only-binary`

### ISSUE THREAD

**Issue by sjincho** [pex-tool/pex#2432]
I'm using pex 2.3.1 on Ubuntu 22.04.

If I create a lock file using:

```
pex3 lock create --output lock.json --only-binary ansicolors ansicolors
```

and run

```
pex --lock lock.json
```

it works w/o problem.

But running:

```
pex --lock lock.json --only-binary ansicolors
```

yields the following error:

```
Resolving wheels was disallowed, but the following project names are configured to only allow resolving pre-built wheels: ansicolors
```

In Pants 2.21.0, it seems like a change that uses pex to handle binary wheels was introduced.
https://github.com/pantsbuild/pants/pull/20598

Due to this change, pants always passes both `--lock` and `--only-binary` flags, causing https://github.com/pantsbuild/pants/issues/21077

**Comment by sjincho** [comment]
Similar issue: https://github.com/pex-tool/pex/issues/2389

**Comment by jsirois** [comment]
The fix for this is now released in Pex 2.4.1: https://github.com/pex-tool/pex/releases/tag/v2.4.1
